sql/planbuilder/ddl.go

```diff
@@ -331,85 +331,136 @@ func assignColumnIndexesInSchema(schema sql.Schema) sql.Schema {
+		pkSch, coll, _ = b.tableSpecToSchema(inScope, outScope, database, strings.ToLower(ct.Table.Name.String()), ct.TableSpec, false)
+	}
+
+	var ok bool
+	var pkOrdinals []int
+	var newSch sql.Schema
+	newSchMap := make(map[string]struct{})
 	var idxDefs sql.IndexDefs
-	if indexableTable, ok := likeTable.Table.(sql.IndexAddressableTable); ok {
-		indexes, err := indexableTable.GetIndexes(b.ctx)
-		if err != nil {
+	var checkDefs []*sql.CheckConstraint
+	for _, likeTable := range ct.OptLike.LikeTables {
+		outScope, ok = b.buildTablescan(outScope, likeTable, nil)
+		if !ok {
+			b.handleErr(sql.ErrTableNotFound.New(likeTable.Name.String()))
+		}
+		lTable, isResTbl := outScope.node.(*plan.ResolvedTable)
+		if !isResTbl {
+			err := fmt.Errorf("expected resolved table: %s", likeTable.Name.String())
 			b.handleErr(err)
 		}
-		for _, index := range indexes {
-			if index.IsGenerated() {
-				continue
-			}
-			constraint := sql.IndexConstraint_None
-			if index.IsUnique() {
-				if index.ID() == "PRIMARY" {
-					constraint = sql.IndexConstraint_Primary
-				} else {
-					constraint = sql.IndexConstraint_Unique
-				}
-			}
 
-			columns := make([]sql.IndexColumn, len(index.Expressions()))
-			for i, col := range index.Expressions() {
-				//TODO: find a better way to get only the column name if the table is present
-				col = strings.TrimPrefix(col, indexableTable.Name()+".")
-				columns[i] = sql.IndexColumn{Name: col}
-			}
-			idxDefs = append(idxDefs, &sql.IndexDef{
-				Name:       index.ID(),
-				Storage:    sql.IndexUsing_Default,
-				Constraint: constraint,
-				Columns:    columns,
-				Comment:    index.Comment(),
-			})
+		if coll == sql.Collation_Unspecified {
+			coll = lTable.Collation()
 		}
-	}
-	origSch := likeTable.Schema()
-	newSch := make(sql.Schema, len(origSch))
-	for i, col := range origSch {
-		tempCol := *col
-		tempCol.Source = newTableName
-		newSch[i] = &tempCol
-	}
 
-	var pkOrdinals []int
-	if pkTable, ok := likeTable.Table.(sql.PrimaryKeyTable); ok {
-		pkOrdinals = pkTable.PrimaryKeySchema().PkOrdinals
-	}
+		if comment == "" {
+			comment = lTable.Comment()
+		}
 
-	var checkDefs []*sql.CheckConstraint
-	if checksTable, ok := likeTable.Table.(sql.CheckTable); ok {
-		checks, err := checksTable.GetChecks(b.ctx)
-		if err != nil {
-			b.handleErr(err)
+		schOff := len(newSch)
+		hasSkippedCols := false
+		for _, col := range lTable.Schema() {
+			newCol := *col
+			name := strings.ToLower(newCol.Name)
+			if _, ok := newSchMap[name]; ok {
+				// TODO: throw warning
+				hasSkippedCols = true
+				continue
+			}
+			newSchMap[name] = struct{}{}
+			newCol.Source = newTableName
+			newSch = append(newSch, &newCol)
 		}
 
-		for _, check := range checks {
-			checkConstraint := b.buildCheckConstraint(outScope, &check)
-			if err != nil {
-				b.handleErr(err)
+		// if a column was skipped due to duplicates, don't copy over PK ords, idxDefs, or checkDefs
+		// since they might be incorrect
+		if hasSkippedCols {
```

Comment:
are the semantics really this simple? seems like you'd need way more testing to figure out correct behavior